File: Demos/Common/CSharp/ITestResponse.cs
using System;
using System.Collections.Generic;
using Ssc.Ssc;
using Ssc.SscAttribute;

public interface ITestResponse {
    [RpcResponse("5F674579-4D3F-42DE-A72C-A8B46AE94908")]
    bool TestResponse(int num, string str, IPeer peer, Action<Action> callback);

    [RpcResponse("1ECE00D8-614A-481F-861E-D20EEA55247C")]
    bool TestResponse(int num, string str, int[] array, IPeer peer, Action<Action> callback);

    [RpcResponse("26D0A8C7-3D9B-4AC9-B6AF-700A61E23BFB")]
    bool TestResponse(int num, string str, int[] array, ITestPacket[] testPackets, IPeer peer, Action<Action> callback);

    [RpcResponse("4844F488-2169-4AAE-A93B-56E45E10495B")]
    void TestResponse(string str, Dictionary<int, ITestPacket> testPackets, IPeer peer, Action<Action> action);
}

File: Demos/Common/Lua/ITestResponse.lua
local classes = require "Classes"

local ITestResponse = classes.class()

function ITestResponse:init(child)
	self.super:init(child)
end

-- User-5F674579-4D3F-42DE-A72C-A8B46AE94908
function ITestResponse:Test1(num,str,peer,callback)
	return true
end

-- User-1ECE00D8-614A-481F-861E-D20EEA55247C
function ITestResponse:Test2(num,str,array,peer,callback)
	return true
end

-- User-26D0A8C7-3D9B-4AC9-B6AF-700A61E23BFB
function ITestResponse:Test3(num,str,array,packets,peer,callback)
	return true
end

-- User-4844F488-2169-4AAE-A93B-56E45E10495B
function ITestResponse:Test4(str,packets,peer,callback)
	return true
end

return ITestResponse